implementations/rust/ockam/ockam_api/src/cli_state/traits.rs

```diff
@@ -8,9 +8,12 @@ use super::Result;
 /// Represents the directory of a type of state and contains
 /// all the data related to that type.
 #[async_trait]
-pub trait StateTrait: Sized {
-    type ItemDir: StateItemDirTrait;
-    type ItemConfig: StateItemConfigTrait + Serialize + for<'a> Deserialize<'a>;
+pub trait StateDirTrait: Sized {
+    type Item: StateItemTrait;
+
+    fn cli_state(&self) -> Result<CliState> {
+        CliState::new(self.dir().parent().expect("no parent dir"))
+    }
 
     fn new(dir: PathBuf) -> Self;
     fn default_filename() -> &'static str;
```

Comment:
I think we should make a clear distinction between:

 - a template / some helper functions to load / persist a `StateDir` and not expose the various `dir` and `path` functions. If there is a `data` dir for a given implementation, the `load` function can do the `std::fs::create_dir_all(dir.join("data"))?;` by itself and maybe use `DATA: &str = "data"` as a constant so we make sure that we are consistent in terms of naming
  
 - the `StateDir` interface which is basically
   - `create(name, config)`
   - `overwrite(name, config)`  
   - `list()`
   - `get(name)`
   - `delete(name)`